### DISCUSSION TITLE
Layer Library

### DISCUSSION THREAD

**Discussion by mattsmith123** [dekart-xyz/dekart#146] [Ideas]
Has there been much discussion about being able to configure layers to be shared across multiple reports?

As I contemplate using Dekart, I can see how I would like to reuse layers (and the datasets they depend on) across multiple reports.  It would also be nice if these layers where "symlinked" so they would update automatically if the common layer was updated.

It seems like it would be pretty straightfoward to store kepler config and dataset config and surface the list in a menu so users could choose it. 

MVP could actually only be configurable via API or config file upload since there would be a lot of value just in letting administrators set common layers that users can select from and use.

I'm thinking about looking into this more and start designing a feature, but want to gauge how much this idea or others like it had already been discussed here.

Thanks!

**Comment by delfrrr** [comment]
@mattsmith123 some thought from my side:

Dekart has an internal concept of the dataset; dataset is pretty much a tab on the right: it can be a query or uploaded file. They are already designed with sharing between users in mind. This should be straight forward to add. I did plan it already, could be accelerated. https://github.com/dekart-xyz/dekart/issues/115

However, I was not thinking about sharing layers. They don't have their own entity in dekart.  I think dataset could have something like "default style" property which would store viz config. Then when the user saves map, you would ask them if they want to update the default style of the dataset 🤔
